core/types/rollup_cost.go

```diff
@@ -103,6 +111,7 @@ func NewRollupCostData(data []byte) (out RollupCostData) {
 			out.ones++
 		}
 	}
+	out.fastlzSize = uint64(FlzCompressLen(data))
```

Comment:
I realize that we now _always_ count zeros and non-zeros, _and_ compute the flzCompressLen. Would it be a large lift to modularize the implementation and only calculate the fields that we need at the time the cost function is applied? Probably something to consider for a follow up. I quickly checked all usages, and we could change the `RollupCostData` struct into an interface with getters for the individual fields, and then lazily calculate them on first access.